### PR TITLE
Issue1993

### LINKED ISSUE #1993: Flow GUI inconsistency
The newly introduced rake selector is able to stay in pretty inconsistent states by typing in an obvious out-of-bound value.
    -- Note 1: the inconsistency isn't corrected even after hitting the "Enter" key
    -- Note 2: the old flow GUI doesn't exhibit the inconsistency.

Screenshot from 2019-12-12 16-11-52

### PR DESCRIPTION
**Pull request by StasJ**
Fix #1993

### CODE REVIEW

File: apps/vaporgui/VDoubleValidator.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <QDoubleValidator>
+
+//! \class VDoubleValidator
+//! Use as a direct replacement of QDoubleValidator.
```
Comment (clyne) [inline apps/vaporgui/VDoubleValidator.h]: Please document how VDoubleValidator differs from QDoubleValidator

Comment (StasJ) [reply apps/vaporgui/VDoubleValidator.h]: done

Comment (clyne) [reply apps/vaporgui/VDoubleValidator.h]: Is there a typo in here:

QDoubleValidator does not accept values outside
//! the range while VDoubleValidator does not allow the user to input
//! values outside of the range

It appears that they do the same thing

**Comment by StasJ** [comment]
> Unfortunately this does not fix #1993. I'm ok with accepting this PR, but the issue remains.
> 
> The QDoubleValidator prohibits its parent from emitting signals when an invalid value is entered. This just leaves the invalid value unchanged until our next round of Update() calls.
> 
> I recommend performing our own validation within the leftTextChanged() and rightTextChanged() functions, and doing away with QValidators. Thoughts?

@pearse, have you tried reproducing #1993?

File: apps/vaporgui/VDoubleValidator.h (hunk 1 above)
Comment (StasJ) [reply apps/vaporgui/VDoubleValidator.h]: QDoubleValidator allows users to input values outside the range.

**Comment by sgpearse** [comment]
Turns out I had a local branch named issue1993, so my test was failing on that branch, not this one.  Seems to work now.

**Pull request merged**